### PR TITLE
Simplify logging

### PR DESCRIPTION
**Pull request by ralphtheninja**
* Closes https://github.com/deltachat/deltachat-desktop/issues/587
* Closes https://github.com/deltachat/deltachat-desktop/issues/585

### CODE REVIEW

**Comment by ralphtheninja** [comment]
I think this makes the logging api easier to use. You can use it as you were using e.g. `console.log` and don't have to think about which parameter does what.

We should also do log rotation but we can tweak this later.

File: src/logger.js [hunk 1]
```diff
@@ -38,72 +36,32 @@ function getStackTrace () {
...
-     */
   constructor (channel) {
     this.channel = channel
   }
 
-  /**
-     * Log a message on **debug** level
-     * @param {string} message the message (human readable)
-     * @param {string} errorCode (optional) machine readable error code (string in snake_case)
-     * @param {string} payload (optional) JSON payload
```
Comment (Jikstra) [inline src/logger.js]: @Simon-Laux likes the method docs for his ide, I'm fine with keeping them.

Comment (ralphtheninja) [reply src/logger.js]: But there are no longer any parameters to document in the logger.

Comment (ralphtheninja) [reply src/logger.js]: Maybe you could be a bit more specific on which method that needs this and i can add it back. I personally dislike this jsdoc style and we don't have it anywhere else in the application. Especially when it means writing 5 lines of documentation for a one line method.

File: src/main/log-handler.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+     * @param {string} ...args Variadic parameters. Stringified before logged to file
+     */
+    log: (channel, level, ...args) => {
+      const timestamp = new Date().toISOString()
+      const out = [
+        timestamp,
+        channel,
+        level
+      ].concat(args.map(arg => JSON.stringify(arg)))
+      stream.write(`${out.join(' ')}\n`)
```
[the remark this answers is not in the document]
Comment (ralphtheninja) [reply src/main/log-handler.js]: > Or do you want to make it space separated? @ralphtheninja

Tab separated is fine.

**Comment by ralphtheninja** [comment]
> The stack-trace should be in a fixed column in the logfile

For DEBUG and INFO this will mean an empty column, which is a bit ugly, but whatever :smile:

File: src/main/developerTools/logHandler.js [hunk 3]
```diff
@@ -1,82 +0,0 @@
...
-
-  if (payload) {
-    var payloadForLogfile = JSON.stringify(payload)
-    if (payloadForLogfile.charAt(0) !== '"') {
-      payloadForLogfile = JSON.stringify(payloadForLogfile)
-    }
-  }
-
-  const logString = `${timestamp}|=|=|"${channel}"|=|=|${level}|=|=|"${message}"|=|=|"${errorCode || ''}"|=|=|${payloadForLogfile || ''}|=|=|${JSON.stringify(stacktrace || '')}`
-    .replace(/\t/g, '  ').replace(/\n|(:?\r\n)/g, '') // Cleanup
```
[the remark this answers is not in the document]
Comment (ralphtheninja) [reply src/main/developerTools/logHandler.js]: > Maybe we should keep this "clean up of the json" to be on the safe side. When we keep it it should be in the map function that converts the args to json.

This is being overly anal imo. For example:

```js
var s = 'wooooooo\tfoobar'
console.log(JSON.stringify(s))
// "wooooooo\tfoobar"
```

Since `JSON.stringify` escapes the `\t`. Same goes for `\n`.

**Comment by Simon-Laux** [comment]
Your right, JSON.stringify should work properly.I woried about the strange results that one get when the string gets stringified multiple times - But thats too unlikely to happen. Am 20.01.2019 23:58 schrieb Lars-Magnus Skog <notifications@github.com>:@ralphtheninja commented on this pull request.



In src/main/developerTools/logHandler.js:
> - * @param {string} errorCode (optional) machine readable error code (string in snake_case)
- * @param {any|string} payload (optional) JSON payload or any other object that can be converted to a JSON string
- * @param {StackFrame} stacktrace (optional) for errors -> gets converted to json
- */
-function log (timestamp, channel, level, message, errorCode, payload, stacktrace) {
-  logDB.push({ channel, level, message, errorCode, payload, stacktrace, timestamp })
-
-  if (payload) {
-    var payloadForLogfile = JSON.stringify(payload)
-    if (payloadForLogfile.charAt(0) !== '"') {
-      payloadForLogfile = JSON.stringify(payloadForLogfile)
-    }
-  }
-
-  const logString = `${timestamp}|=|=|"${channel}"|=|=|${level}|=|=|"${message}"|=|=|"${errorCode || ''}"|=|=|${payloadForLogfile || ''}|=|=|${JSON.stringify(stacktrace || '')}`
-    .replace(/\t/g, '  ').replace(/\n|(:?\r\n)/g, '') // Cleanup


Maybe we should keep this "clean up of the json" to be on the safe side. When we keep it it should be in the map function that converts the args to json.

This is being overly anal imo. For example:
var s = 'wooooooo\tfoobar'
console.log(JSON.stringify(s))
// "wooooooo\tfoobar"
Since JSON.stringify escapes the \t. Same goes for \n.

**Comment by Jikstra** [comment]
@Simon-Laux can you fix your last message? The code blocks make it super messy/hard to parse ^^

**Comment by Simon-Laux** [comment]
> @Simon-Laux can you fix your last message? The code blocks make it super messy/hard to parse ^^

Gmail and Github messed up my reply...

File: src/main/developerTools/logHandler.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Simon-Laux) [reply src/main/developerTools/logHandler.js]: Your right, JSON.stringify should work properly.I woried about the strange results that one get when the string gets stringified multiple times - But thats too unlikely to happen.
